### PR TITLE
feat: implement notification controller on audio recitation 

### PR DESCRIPTION
**Pull request by yumnapratista24**
### Background

- Add functionality to control audio recitation from notification settings
- Wrap audio player with audio handler to be able to use notification control given by audio_service package

### Screenshot

https://github.com/Yaumi-Digital-School/quranplus-flutter/assets/36220334/868cd14b-15cf-49e7-9f74-1fe48cc607fe

https://github.com/Yaumi-Digital-School/quranplus-flutter/assets/36220334/240a63f3-c3f9-4512-8769-8a051ac77106

### Impacted Products

- Surah page
- Notification

### How to Test

- Tested locally on iOS and android

### Pre-Deployment Checklist

- [ ] Run Local OK (Mandatory)
- [x] Run Dev Build on Real Device

### Test Checklist

- [ ] Unit Testing
- [ ] Other (Please Elaborate)

### Code Review Guidelines

https://oyteam.quip.com/4m8kAafMWWiX/Code-Review-Manifesto

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -3,13 +3,15 @@
     package="com.yaumi.qurantafsir.id">
     
     <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" tools:node="remove"/>
+    <uses-permission android:name="android.permission.WAKE_LOCK"/>
```
Comment (indraseptama) [inline android/app/src/main/AndroidManifest.xml:6]: currious, ini berarti ngijinin screen buat selalu awake kan ya yum(?), dan berarti emang diperlukan ya untuk fitur ini?

File: lib/pages/home_page_v2/home_page_state_notifier.dart [hunk 2]
```diff
@@ -138,15 +133,13 @@ class HomePageStateNotifier extends BaseStateNotifier<HomePageState> {
     required Function() onPlayBackError,
   }) async {
     _setAudioSuratLoaded(surat);
-    await audioPlayerNotifier.init(
-      AudioBottomSheetState(
+    await _audioRecitationNotifier.init(
```
Comment (indraseptama) [inline lib/pages/home_page_v2/home_page_state_notifier.dart:136]: Nice! It looks simpler!

File: lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  Future<void> previousSurah() async {
+    final int currentSurahId = state.surahId;
+    final int previousSurahId = currentSurahId - 1;
+    await _startNewSurah(previousSurahId);
+  }
+}
+
+final audioBottomSheetProvider =
```
Comment (indraseptama) [inline lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart:208]: ini diganti juga ga sih yum harusnya? Biar konsisten. "audioRecitationProvider" mungkin

File: lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        MediaItem(
+          id: '${state.surahId}-${state.ayahId}-$reciterId',
+          title: '${state.surahName} - Ayat: ${state.ayahId}',
+          // replace with dynamic reciter name
+          artist: 'Mishari Rashid al-`Afasy',
+          extras: <String, dynamic>{
+            'url': response.data.audioFileUrl,
+          },
+        ),
+      );
```
Comment (indraseptama) [inline lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart:84]: salah satu kuncinya disini yak yum berarti? Ini udah include set url yak? Gokil. Explain singkat dong yum

**Review by indraseptama — APPROVED** [review]
approved with minor question

**Review by jokot — APPROVED** [review]

File: lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart (hunk 4 above)
Comment (yumnapratista24) [reply lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart:84]: iya ini kunci utk enable notification nya. Nah tp sebenernya play url nya masih manual ndras, jd di dalam function `setMediaItem` dari audiohandler itu, gue panggil lagi utk set url dari si extras ini

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (yumnapratista24) [reply android/app/src/main/AndroidManifest.xml:6]: wah benar ndras, hrsnya ini gak perlu.. tdnya gua berasumsi ini dibutuhin krn kita klo jalan di background mesti hit API kan ketika pindah ayat, ternyata bener fokus utamanya ngijinin screen selalu awake. Gua delete yak, thanks a lot ndras!

File: lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart (hunk 3 above)
Comment (yumnapratista24) [reply lib/widgets/audio_bottom_sheet/audio_recitation_state_notifier.dart:208]: updated ndras

**Pull request merged**